src/nvim/ex_getln.c

```diff
@@ -2398,13 +2398,25 @@ static void abandon_cmdline(void)
 /// @param indent  indent for inside conditionals
 char_u *getcmdline(int firstc, long count, int indent, bool do_concat FUNC_ATTR_UNUSED)
 {
+  bool check_cmdheight = p_ch < 1 && !ui_has(kUIMessages);
+  long save_cmdheight = p_ch;
+
+  if (check_cmdheight) {
+    set_option_value("ch", 1L, NULL, 0);
```

Comment:
I don't like this. it would be better to check where `p_ch` is used if it needs to be adjusted, provided it is not too many places.

Reply:
I know it is better, but `p_ch` is used in 68 places.
It is not so many?